src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java

```diff
@@ -147,6 +148,8 @@
     private static final int TIME_TO_LIVE = JMeterUtils.getPropDefault("httpclient4.time_to_live", 2000);
 
     private static final String CONTEXT_METRICS = "jmeter_metrics"; // TODO hack for metrics related to HTTPCLIENT-1081, to be removed later
+    
+    private static final Pattern PORT_PATTERN = Pattern.compile("^\\d+$");
```

Comment:
I'm sure you are aware of this, but let's mention it anyway: Pattern is not thread-safe, so you have to be careful with static instances. (Of course I lack the necessary knowledge of jmeter to judge wether it's save to reuse the pattern in these classes.)